dts/bindings/clock/nordic,nrf-global-hsfll.yaml

```diff
@@ -0,0 +1,37 @@
+# Copyright (c) 2024 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  Nordic Global HSFLL clock.
+
+  The lowest supported clock frequency is the default
+  clock frequency.
+
+  Example:
+
+    global_hsfll: global_hsfll {
+            compatible = "nordic,nrf-global-hsfll";
+            clocks = <&fll16>;
+            #clock-cells = <0>;
+            supported-clock-frequencies = <64000000
+                                           128000000
+                                           256000000
+                                           320000000>;
+    };
+
+compatible: "nordic,nrf-global-hsfll"
+
+include:
+  - "base.yaml"
+  - "clock-controller.yaml"
+
+properties:
+  clocks:
+    required: true
+
+  "#clock-cells":
+    const: 0
+
+  supported-clock-frequencies:
+    type: array
+    description: Supported clock frequencies in ascending order
```

Comment:
Usually, the supported values are only described as an enum in the `clock-frequency` property binding. Hypothetically, if there was a DT macro for retrieving all possible enum values, would it make sense to use that in the driver, instead of inventing a new property?

Reply:
This property is distinct from `clock-frequency` given that the actual frequency is not known, it will be one of the supported frequencies, and it will vary. The user can't select any of them meaningfully in the devicetree, only describe which the driver can request, hence why I added this :)

Reply:
It has no `clock-frequency` binding since its not a fixed clock :)

Reply:
That makes sense. I guess `clock-frequency` could still signify the frequency requested at boot, but I see that Kconfig is used for that, so I assume that's more appropriate.